components/telemetry-fluentbit-sidecar/main.go

```diff
@@ -0,0 +1,126 @@
+package main
+
+import (
+	"io/ioutil"
+	"net/http"
+	"os"
+	"path/filepath"
+	"strconv"
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+)
+
+type directory struct {
+	name string
+	size int64
+}
+
+var (
+	fsBuffeLabelsVector = promauto.NewGaugeVec(prometheus.GaugeOpts{
+		Namespace: "",
+		Subsystem: "",
+		Name:      "telemetry_fsbuffer_vector",
+		Help:      "Disk size for different emitters",
+	}, []string{"name"})
+
+	fsbufferSize = promauto.NewGauge(prometheus.GaugeOpts{
+		Namespace:   "",
+		Subsystem:   "",
+		Name:        "telemetry_fsbuffer_size",
+		Help:        "The total disk size of the fluentbit chunk buffer",
+		ConstLabels: nil,
+	})
+
+	fsbufferMax = promauto.NewGauge(prometheus.GaugeOpts{
+		Namespace:   "",
+		Subsystem:   "",
+		Name:        "telemetry_fsbuffer_limit",
+		Help:        "The maximum defined size of the fluentbit chunk buffer",
+		ConstLabels: nil,
+	})
+)
+
+func main() {
```

Comment:
Let's have some minimal logging that at least initially prints the watched directory and served port. In best case, you would also have an info log level that logs every iteration. Consider using [this](https://github.com/kyma-project/kyma/tree/main/common/logging/logger) logging library.